static/app/stories/view/storyHeading.tsx

```diff
@@ -0,0 +1,76 @@
+import type {ComponentProps} from 'react';
+import {Fragment} from 'react';
+import {flushSync} from 'react-dom';
+import {createRoot} from 'react-dom/client';
+import styled from '@emotion/styled';
+
+import {LinkButton} from 'sentry/components/core/button/linkButton';
+import {Flex} from 'sentry/components/core/layout';
+import {Heading} from 'sentry/components/core/text';
+import {IconLink} from 'sentry/icons';
+import {useStory} from 'sentry/stories/view/useStory';
+import slugify from 'sentry/utils/slugify';
+import useCopyToClipboard from 'sentry/utils/useCopyToClipboard';
+
+export function StoryHeading(props: ComponentProps<typeof Heading>) {
+  const {story} = useStory();
+  const storyTitle = (story.exports.frontmatter as any)?.title;
+  const text = stringifyChildren(props.children);
+  const id = props.id ?? slugify(text);
+  const {onClick} = useCopyToClipboard({
+    text: `${window.location.toString().replace(/#.*$/, '')}#${id}`,
+    successMessage: (
+      <Fragment>
+        Copied link to{' '}
+        <strong>
+          {storyTitle ? `${storyTitle} > ` : null}
+          {text}
+        </strong>
+      </Fragment>
+    ),
+  });
+
+  return (
+    <HeadingContainer gap="md" align="center">
+      <Heading {...props} id={id} />
+      <LinkButton priority="transparent" size="xs" href={`#${id}`} onClick={onClick}>
+        <IconLink />
+      </LinkButton>
+    </HeadingContainer>
+  );
+}
+
+const HeadingContainer = styled(Flex)`
```

Comment:
it’s a bit sad that we need to style our own `Flex` like that. Maybe we can improve this in a follow-up?